windows-10-setup.md

```diff
@@ -7,45 +10,132 @@
 1. Install the following VS Code extensions
     - ESLint
-    - WSL - Remote
+    - Remote - WSL
     - Live Share (online students only)
     - vscode-icons (optional, but pretty :wink:)
     - Bracket Pair Colorizer (optional)
     - GitLens (optional)
 1. Restart your terminal
 1. Open VS Code again with `code .`
     - This should begin downloading the VS Code Server
-    - When prompted for access, accept it
+    - When prompted for access (by Windows Defender Firewall), click "Allow access"
 1. Install nvm
-    - `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
+    - Enter this command into your terminal: `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
 1. Move the 3 nvm lines from the bottom of `.bashrc` to the bottom of `.zshrc`
+    - This is the trickiest step! 
+    - Run this command to open your .bashrc config file: `code ~/.bashrc`
+    - Scroll down to the bottom of the file and cut the three lines at the bottom that look like this: 
+
+    ```j
+        export NVM_DIR="$HOME/.nvm"
+        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+        [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+    ```
+
+    - Run this command to open your .zshrc file: `code ~/.zshrc`
+    - Paste the nvm lines you cut from the .bashrc file down at the bottom of the .zshrc file
+1. Get rid of the distracting green highlighting in your terminal (optional, but recommended):
+    - In your .zshrc file (that you opened in the above step), find this line and uncomment it (i.e. delete the `#`):
+    ```
+    # DISABLE LS_COLORS="true"
+    ```
 1. Change the oh-my-zsh theme to 'bira'
+    - Scroll to the top of the .zshrc file and replace the `ZSH_THEME=` value with 'bira'
 1. Restart your terminal
-1. `nvm install --lts`
-
-1. Enable the automatic fixing of linting errors on file save by adding this to your `settings.json`
+1. Install the latest version of nvm 
+    - Run this command in your terminal: `nvm install --lts`
+1. In your Windows Terminal install some global npm packages
+    - Copy/paste in this entire chunk of code (if you get a warning about multiple lines of text, click 'Paste anyway'):
+ 
+    ```sh
+    npm install -g \
+    eslint@5.16.0 \
+    babel-eslint@10.0.1 \
+    eslint-config-standard@12.0.0 \
+    eslint-plugin-import@2.17.3 \
+    eslint-plugin-node@9.1.0 \
+    eslint-plugin-promise@4.1.1 \
+    eslint-plugin-react@7.13.0 \
+    eslint-plugin-standard@4.0.0
+    ```
+1. Make VS Code your default Git editor
+    - Run this command in your terminal: `git config --global core.editor "code --wait"`
+1. Enable the automatic fixing of linting errors on file save by adding this to your `settings.json` in VS Code:
     ```js
     "editor.codeActionsOnSave": {
         "source.fixAll.eslint": true
     }
     ```
+1. Restart your PC
+
+## Where to save your files
```

Comment:
We should keep checking on the validity of this section. The WSL team at Microsoft are continuing to make improvements to the filesystem integration between Linux and Windows. Over time this will become less of an issue. I'm personally not sure at this point how much of this is out of date or if any of it is.

Reply:
This "Where to store your files" section is written by me based on having moved files from Windows to Linux on a student's PC and also my own, a few weeks back. So I can vouch for the information about filepaths etc. being *current*, and it worked well for me and I believe also for the student... but I'm unsure whether it's the right answer for Foundations. Like you say though, hopefully Microsoft might improve the integration in the future, or maybe there's some trick that already exists that I'm not aware of.